src/value.cpp

```diff
@@ -678,7 +678,7 @@ Expr MemRef::storeArray(
 Expr MemRef::isInBounds() const {
   auto numelem = m->getNumElementsOfMemBlock(bid);
   auto memrefSize = get1DSize();
-  return numelem.uge(memrefSize) & ((Expr)offset).ult(numelem - memrefSize);
+  return numelem.uge(memrefSize) & ((Expr)offset).ule(numelem - memrefSize);
```

Comment:
This might be undiscovered bugs in `isInBounds` encoding.
Offset can be zero if `numelem` and `memrefSize` has same value.. but in previous `ult` encoding this was not allowed.